docs/source/cpp/compute.rst

```diff
@@ -575,6 +575,22 @@ when a positive ``max_splits`` is given.
   as separator.
 
 
+String joining
+~~~~~~~~~~~~~~
+
+The inverse of string splitting:
+
++--------------------------+------------+-------------------------+-------------------+---------+
+| Function name            | Arity      | Input types             | Output type       | Notes   |
++==========================+============+=========================+===================+=========+
+| binary_join              | Binary     | List-like               | String-like       | \(1)    |
```

Comment:
Since the two inputs must be of different types, it would be nice to make this explicit here (perhaps by making two columns "first input type" and "second input type"?).